zp-relayer/pool.ts

```diff
@@ -154,12 +154,12 @@ class Pool {
     }
 
     const lastBlockNumber = await this.getLastBlockToProcess()
-    let finishBlock = fromBlock
-    for (let startBlock = fromBlock; finishBlock <= lastBlockNumber; startBlock = finishBlock) {
-      finishBlock += config.eventsProcessingBatchSize
+    let toBlock = startBlock
+    for (let fromBlock = startBlock; toBlock <= lastBlockNumber + 1; startBlock = toBlock) {
+      toBlock += config.eventsProcessingBatchSize
```

Comment:
Why we don't take minimum of `toBlock` and `lastBlockNumber + 1` here?

Reply:
Actually, we can. But, why is this important?

Reply:
This means we can process same event at the end of the interval multiple times, should be extra careful that nothing breaks (i.e. all state modifications) on such repeated processing

Reply:
>This means we can process same event at the end of the interval multiple times

Why?

Anyway, all state updates in `syncState` loop are idempotent, so nothing would break if same event would be processed twice.
The original issue, was that range event queries included `BATCH_SIZE+1` elements and RPC rejected it.